eras/shelley/impl/src/Cardano/Ledger/Shelley/Genesis.hs

```diff
@@ -129,6 +149,71 @@ emptyGenesisStaking =
       sgsStake = mempty
     }
 
+-- | Unlike @'NominalDiffTime'@ that supports @'Pico'@ precision, this type
+-- only supports @'Micro'@ precision.
+newtype NominalDiffTimeMicro = NominalDiffTimeMicro Micro
+  deriving (Show, Eq, Generic)
+  deriving anyclass (NoThunks)
+  deriving newtype (Ord, Num, Fractional, Real, ToJSON, FromJSON)
+
+deriving instance Generic Micro
+
+deriving anyclass instance NoThunks Micro
+
+instance ToCBOR NominalDiffTimeMicro where
+  toCBOR = encodeNominalDiffTimeMicro
```

Comment:
In fact the `To/FromCBOR` instances can be derived using GND. All you need is to add these standalone instances to:

`Cardano.Ledger.Binary.Encoding.ToCBOR` and `Cardano.Ledger.Binary.Decoding.FromCBOR`:

```haskell
deriving newtype instance ToCBOR (MkFixed p)
deriving newtype instance FromCBOR (MkFixed p)
```

Reply:
I found instances already present for `Nano`, `Pico`, so I added similar ones directly for `Micro` instead of deriving. Is that okay?

Reply:
I have replaced all instances (`Nano`, `Pico`, `Micro`) with `deriving newtype instance Typeable p => {To|From}CBOR (Fixed p)`